test/Dialect/ESI/bundles.mlir

```diff
@@ -0,0 +1,36 @@
+// RUN: circt-opt %s | circt-opt | FileCheck %s
+
+!bundleType = !esi.bundle<[
+  !esi.channel<i8> to "data",
+  !esi.channel<none> from ack]>
+
+// CHECK-LABEL: hw.module @Receiver(%foo: !esi.bundle<[!esi.channel<i8> to "data", !esi.channel<none> from "ack"]>) {
+// CHECK-NEXT:     [[R0:%.+]] = esi.null : !esi.channel<none>
+// CHECK-NEXT:     [[R1:%.+]] = esi.bundle.unpack [[R0]] from %foo : !esi.bundle<[!esi.channel<i8> to "data", !esi.channel<none> from "ack"]>
+hw.module @Receiver(%foo: !bundleType) {
+  %ack = esi.null : !esi.channel<none>
+  %data = esi.bundle.unpack %ack from %foo : !bundleType
```

Comment:
I would also output `%data` here from the `hw.module` which implicitly will check that `%data` indeed is the expected type (since you don't have a verifier).